src/utils/internal.ts

```diff
@@ -528,7 +528,11 @@ export async function getApiAtBlock(
 
 type QueryMultiParam<T extends AugmentedQuery<'promise', AnyFunction>[]> = {
   [index in keyof T]: T[index] extends AugmentedQuery<'promise', infer Fun>
-    ? [T[index], ...Parameters<Fun>]
+    ? Fun extends (firstArg: infer First, ...restArg: infer Rest) => ReturnType<Fun>
```

Comment:
Nice work, I like the API this provides. Even if its a bit hard to read people shouldn't really have to look at it.